pkg/api/api.go

```diff
@@ -89,6 +89,11 @@ func (hs *HTTPServer) registerRoutes() {
 	r.Get("/a/:id/*", reqSignedIn, hs.Index) // App Root Page
 	r.Get("/a/:id", reqSignedIn, hs.Index)
 
+	//pubdash
+	if hs.Features.IsEnabled(featuremgmt.FlagPublicDashboards) {
+		r.Get("/public-dashboards/:uid", redirectFromLegacyPanelEditURL, hs.Index)
```

Comment:
This needs cannot go to Index directly, it needs a security middleware as well that validates the uid exists, but can be added later

Reply:
Loading a public dashboard with url that does not exist can return 404 would be the main reason. 


But it's not super important, at least for an mvp . We can validate the url in the frontend and show not found error there.